src/contracts/libraries/GPv2Encoding.sol

```diff
@@ -0,0 +1,119 @@
+// SPDX-license-identifier: LGPL-3.0-or-newer
+pragma solidity ^0.6.12;
+
+import "@openzeppelin/contracts/token/ERC20/IERC20.sol";
+
+/// @title Gnosis Protocol v2 Encoding Library.
+/// @author Gnosis Developers
+library GPv2Encoding {
+    /// @dev A struct representing an executed order that can be used for
+    /// settling a batch.
+    struct Order {
+        address owner;
+        IERC20 sellToken;
+        IERC20 buyToken;
+        uint256 sellAmount;
+        uint256 buyAmount;
+        uint32 validTo;
+        uint256 tip;
+        OrderKind kind;
+        OrderFill fill;
+        uint256 executedAmount;
+    }
+
+    /// @dev An enum describing an order kind, either a buy or a sell order.
+    enum OrderKind {Sell, Buy}
```

Comment:
Will the compile compress this into a single bit in terms of memory footprint?

Reply:
No, struct memory is not packed.

However, if we only ever allocate memory for a single order, then use it right away, then we would just pay for the additional word of memory (for splitting the `uint8` flags into `OrderKind` and `bool`). I think the extra cost will be negligible then.